tests/trace_server/conftest.py

```diff
@@ -25,7 +25,7 @@ def pytest_addoption(parser):
             "--trace-server",
             action="store",
             default="clickhouse",
-            help="Specify the client object to use: sqlite or clickhouse",
+            help="Specify the client object to use: sqlite, clickhouse, or stainless",
```

Comment:
I don't think this is controlling the correct thing. This flag changes the server engine. However, you have used it in the `remote_http_trace_server` fixture which is used in the `trace_server` fixture which was added recently and overloads the existing `trace_server` fixture. I think we should:
1. Rename the fixtures so they are not in conflict
2. Don't re-use this flag - instead create a new one that doesn't overload existing stuff